### PR TITLE
adds with_count to fetch_guild

### LINKED ISSUE #159: Fetch_guild getting with_counts
### Summary

This basically adds the ability to correctly get prescence_count

### What is the feature request for?

The core library

### The Problem

The problem is that you can't do this on the guild object itself(and to add stuff that doesn't exist in guild stuff)

### The Ideal Solution

I would like this fork to be able to get prescence_count, the new member_count at the time, and of course other guild information

having a fetch_guild with the prescence_count and member_count as a part of the guild object, otherwise normal get would have prescence_count None, and member_count would be the most updated(it also doesn't require intents to get all the users online, you can't always get intents either. Discord would probaly like it even more if you added this)
and being able to get with_counts on many guilds with fetch_guilds.

### The Current Solution

The current one is to fetch the invite(which as we know we don't always have permissions to make, and not all guilds have vanity urls), but even so you should only have to be able to use fetch_guild or fetch_guilds.

for example now you need to have an invite object to get the prescence count, and an accurate member_count.

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by JDJGInc**
## Summary

Closes #159

## Checklist

- [x] If code changes were made then they have been tested.
    - [x] I have updated the documentation to reflect the changes.
- [ ] If `type: ignore` comments were used, a comment is also left explaining why
- [ ] This PR fixes an issue.
- [x] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, typehinting, examples, ...)

### CODE REVIEW

**Comment by JDJGInc** [comment]
I have tested this, do you want me to swap it to uh

**Comment by JDJGInc** [comment]
so wait_counts is OFf by Default or are you fine with this?

**Comment by JDJGInc** [comment]
any like typing checks at all?

**Comment by JDJGInc** [comment]
import discord
client = discord.Client(intents = discord.Intents.all())

@client.event
async def on_ready():
    print("client is ready")

    guild = await client.fetch_guild(guild_id)
    print(guild.approximate_member_count, guild.approximate_presence_count)

    guild = await client.fetch_guild(guild_id, with_counts = False)
    print(guild.approximate_member_count, guild.approximate_presence_count)

TOKEN = ""
client.run(TOKEN)

**Comment by JDJGInc** [comment]
Returns:
client is ready
522 297

**Comment by JDJGInc** [comment]
Yes I know the test code is weird, but it works

**Comment by Lulalaby** [comment]
Please do not spam comments.
Try to put everything in one.
Too many comments spams our discord GitHub feed Channel.

Furthermore I just assigned the needed things. You have to wait for reviewers.

For now this is fine, we come back with feedback later.

**Review by Dorukyum — COMMENTED** [review]
Perhaps you should add `versionadded` to the docstring

**Comment by JDJGInc** [comment]
> Perhaps you should add `versionadded` to the docstring

is this version 2.0.0?

**Comment by Dorukyum** [comment]
> > Perhaps you should add `versionadded` to the docstring
> 
> is this version 2.0.0?

I guess so.

**Comment by JDJGInc** [comment]
> > > Perhaps you should add `versionadded` to the docstring
> > 
> > 
> > is this version 2.0.0?
> 
> I guess so.

this change should work, I am just waiting on checks to finish.

**Review by Lulalaby — CHANGES_REQUESTED** [review]
Inconsistent.
Please fix newlines and spaces

File: discord/guild.py [hunk 1]
```diff
@@ -253,6 +253,18 @@ class Guild(Hashable):
         The guild's NSFW level.
 
         .. versionadded:: 2.0
+        
+         approximate_member_count: Optional[:class:`int`]
```
Comment (Dorukyum) [inline discord/guild.py:257]:
```suggestion
    approximate_member_count: Optional[:class:`int`]
```

File: discord/guild.py [hunk 2]
```diff
@@ -296,6 +308,8 @@ class Guild(Hashable):
         '_public_updates_channel_id',
         '_stage_instances',
         '_threads',
+        "approximate_member_count",
+        "approximate_presence_count"
```
Comment (Dorukyum) [inline discord/guild.py:312]:
```suggestion
        "approximate_presence_count",
```

File: discord/guild.py [hunk 3]
```diff
@@ -253,6 +253,18 @@ class Guild(Hashable):
         The guild's NSFW level.
 
         .. versionadded:: 2.0
+        
+    approximate_member_count: Optional[:class:`int`]
+        The approximate number of members in the guild. This is ``None`` unless the guild is obtained
+        using :meth:`Client.fetch_guild` with ``with_counts=True``.
+        .. versionadded:: 2.0
```
Comment (Lulalaby) [inline discord/guild.py:260]:
```suggestion

        .. versionadded:: 2.0
```

**Comment by Dorukyum** [comment]
One last thing: there should be a newline before `versionadded`s

File: discord/guild.py [hunk 4]
```diff
@@ -253,6 +253,18 @@ class Guild(Hashable):
...
+        
+    approximate_presence_count: Optional[:class:`int`]
+        The approximate number of members currently active in the guild.
+        This includes idle, dnd, online, and invisible members. Offline members are excluded.
+        This is ``None`` unless the guild is obtained using :meth:`Client.fetch_guild`
+        with ``with_counts=True``.
+        .. versionadded:: 2.0
```
Comment (Lulalaby) [inline discord/guild.py:267]:
```suggestion

        .. versionadded:: 2.0
```

**Comment by JDJGInc** [comment]
> Inconsistent. Please fix newlines and spaces

I have fixed everything I have seen, but I don't see the other fixes you want me to make.

File: discord/client.py [hunk 5]
```diff
@@ -1215,6 +1215,13 @@ async def fetch_guild(self, guild_id: int, /) -> Guild:
         guild_id: :class:`int`
             The guild's ID to fetch from.
 
+        with_counts: :class:`bool`
+            Whether to include count information in the guild. This fills the
+            :attr:`.Guild.approximate_member_count` and :attr:`.Guild.approximate_presence_count`
+            fields.
+            .. versionadded:: 2.0
+
```
Comment (Lulalaby) [inline discord/client.py:1223]:
```suggestion

            .. versionadded:: 2.0
```

File: discord/client.py [hunk 6]
```diff
@@ -1215,6 +1215,13 @@ async def fetch_guild(self, guild_id: int, /) -> Guild:
...
+
```
Comment (Lulalaby) [inline discord/client.py:1224]:
```suggestion
```
Remove line

File: discord/guild.py (hunk 1 above)
Comment (JDJGInc) [reply discord/guild.py]: fixed

File: discord/guild.py (hunk 2 above)
Comment (JDJGInc) [reply discord/guild.py]: also fixed

**Review by Lulalaby — CHANGES_REQUESTED** [review]

**Review by JDJGInc — COMMENTED** [review]
what else do I need to fix here?

**Comment by JDJGInc** [comment]
I have merged all the changes :D

**Comment by JDJGInc** [comment]
how is change still requested, when I committed all the changes you wanted?

**Comment by JDJGInc** [comment]
Scrutinize just canceled itself?

**Review by Dorukyum — APPROVED** [review]

**Comment by BobDotCom** [comment]
> Scrutinize just canceled itself?

It does when you push commits too close to each other. I've re-queued it.

**Comment by JDJGInc** [comment]
> > Scrutinize just canceled itself?
> 
> It does when you push commits too close to each other. I've re-queued it.

Passed all checks :D

**Review by Lulalaby — APPROVED** [review]

**Pull request merged**